pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -102,6 +102,13 @@ type WorkflowSpec struct {
 	// ServiceAccountName is the name of the ServiceAccount to run all pods of the workflow as.
 	ServiceAccountName string `json:"serviceAccountName,omitempty"`
 
+	// AutomountServiceAccountToken indicates whether a service account token should be automatically mounted in pods.
+	// ExecutorServiceAccountTokenName must be specified if this value is false.
+	AutomountServiceAccountToken *bool `json:"automountServiceAccountToken,omitempty"`
+
+	// ExecutorServiceAccountTokenName specifies a service account token name for executor containers.
+	ExecutorServiceAccountTokenName string `json:"executorServiceAccountTokenName,omitempty"`
```

Comment:
@dtaniwaki - i was thinking, instead of `executorServiceAccountTokenName`, should this reference a service account instead and we would have the controller look up the the secret name from the service account for the volume mount?

This seems like it would be more user-friendly since users do not generally create tokens, they create service accounts. Note that this would require expanding the privileges of the controller to have read permissions on service accounts.

Reply:
Agree. I will change it.

Reply:
If we use `ExecutorServiceAccountName` instead of `ExecutorServiceAccountTokenName`, do you think `AutomountServiceAccountToken` still make sense to only disable automount service account token on the main container?

Reply:
FYI
Found the corresponded code in Kubernetes
https://github.com/kubernetes/kubernetes/blob/b8eecd671d7d895ecd4f0fc6930b82e55298e257/plugin/pkg/admission/serviceaccount/admission.go#L422-L437

Reply:
> If we use ExecutorServiceAccountName instead of ExecutorServiceAccountTokenName, do you think AutomountServiceAccountToken still make sense to only disable automount service account token on the main container?

Yes. I think the following should disable mounting of service accounts entirely (in both main and wait):
```yaml
spec:
  automountServiceAccountToken: false
```

and the following will disable it on main, but mount a different service account for the executor:

```yaml
spec:
  automountServiceAccountToken: false
  executorServiceAccountName: foo
```

And the following would use separate service accounts for main and wait
```yaml
spec:
  serviceAccountName: bar
  executorServiceAccountName: foo
```

Reply:
Sounds good.
I realized we need to add read permission for service accounts to the cluster role.

Reply:
For the following case:

```
spec:
  automountServiceAccountToken: false
```

We can't run a workflow because the current executor needs a service account token during the initialization, so I made it cause an error during validation.